holidays/countries/niue.py

```diff
@@ -0,0 +1,109 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays import SEP
+from holidays.groups import ChristianHolidays, InternationalHolidays, StaticHolidays
+from holidays.observed_holiday_base import ObservedHolidayBase, SAT_SUN_TO_NEXT_MON_TUE
+
+
+class Niue(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, StaticHolidays):
+    """Niue holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Niue>
+        * <https://en.wikipedia.org/wiki/Niue_Constitution_Act>
+        * [Public Holidays Ordinance 1961](https://web.archive.org/web/20250102100637/http://www.paclii.org/nu/legis/num_act/nipho1961314.pdf)
+        * <https://www.gov.nu/public-service-circulars>
+        * <http://archive.today/2025.07.14-145535/https://www.wipo.int/wipolex/en/text/427817>
+        * <https://web.archive.org/web/20250223114854/https://niuepocketguide.com/public-holidays-in-niue-other-important-dates/>
+        * [Queen's Birthday](https://web.archive.org/web/20220928145826/https://publicholidays.asia/niue/queens-birthday/)
+        * [2018](https://web.archive.org/web/20230322165612/https://publicholidays.asia/niue/2018-dates/)
+        * [2025](https://web.archive.org/web/20241208034202/https://www.qppstudio.net/publicholidays2025/niue.htm)
```

Comment:
```suggestion
        * <https://web.archive.org/web/20250223114854/https://niuepocketguide.com/public-holidays-in-niue-other-important-dates/>
        * [2021/2022 Public Holidays](https://web.archive.org/web/20250509105501/https://www.gov.nu/media/pages/information/1018c58017-1725838374/4nov2021-circular.pdf)
        * [2025 Good Friday & Easter Monday and Anzac Day](https://web.archive.org/web/20250719195126/https://www.gov.nu/media/pages/gazette/3e02b3aa84-1746400484/official-psc-circular-easter-anzac-holiday.pdf)
        * [2025 King's Birthday](https://web.archive.org/web/20250719195301/https://www.gov.nu/media/pages/public-service-circulars/3b9f44b6a0-1748573974/king-s-birthday-public-holiday-2025-circular.pdf)
        * <http://archive.today/2025.07.14-145535/https://www.wipo.int/wipolex/en/text/427817>
```
I added links to specific documents from Public Service Circulars (please correct me if I missed something) and removed aggregators. Also, I didn't quite realize usefulness of the last link.

Reply:
So the last link had a little section that basically told that holidays follow the `SAT_SUN_TO_MON_TUE`, so I thought of adding it here since some non-gov websites just say: `Note that if any public holiday dates fall on a weekend, the next Monday is observed as a national public holiday too.`

<img width="840" height="311" alt="image" src="https://github.com/user-attachments/assets/47dc7e34-713c-4d17-a686-02abc74fb693" />